### PR TITLE
Game form

### PR DESCRIPTION
**Pull request by DeepakChowdary0811**
Floating button to create a form for the game and its validation, submission, adding to database is done.

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -1453,4 +1453,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 53c342e88de2c9ccb956b36de9ad5841aeded8fb
 
-COCOAPODS: 1.16.2
+COCOAPODS: 1.15.2
```
Comment (jossean-rivera) [inline ios/Podfile.lock:1456]: Not sure we should do this.
```suggestion
COCOAPODS: 1.16.2
```

File: lib/navigation/route_names.dart [hunk 2]
```diff
@@ -10,4 +10,5 @@ class RouteNames {
   static const String historyDetails = "historyDetails";
 
   static String registrationConfirmation = "registration-confirmation";
+  static String gameForm = 'gameForm';
```
Comment (jossean-rivera) [inline lib/navigation/route_names.dart:13]: I'd add const
```suggestion
  static const String gameForm = 'gameForm';
```

File: lib/pages/sport_details_page.dart [hunk 3]
```diff
@@ -24,7 +24,15 @@ class SportDetailsPage extends StatefulWidget {
 }
 
 class _SportDetailsPageState extends State<SportDetailsPage> {
-  bool showUnavailable = false; // Toggle for available/unavailable events
+  bool showUnavailable = false;
+  
+    void _navigateToForm(BuildContext context, String sportId) {
+    print(sportId);
```
Comment (jossean-rivera) [inline lib/pages/sport_details_page.dart:30]:
```suggestion
```

**Comment by DeepakChowdary0811** [comment]
All the changes mentioned have been made.

**Comment by jossean-rivera** [comment]
> All the changes mentioned have been made.

Thanks will merge this.

File: ios/build/ios/XCBuildData/PIFCache/target/TARGET@v11_hash=39047dbfca6f81dc03f9b20d0e79586f-json [hunk 4]
```diff
@@ -0,0 +1 @@
...eference":"bfdfe7dc352907fc980b868725387e9890f568fc9b811cd56b08e401eec0f35e"}],"guid":"bfdfe7dc352907fc980b868725387e981fc8e779d69bcb3e6b4cbb0e393c16a6","type":"com.apple.buildphase.resources"}],"buildRules":[],"dependencies":[{"guid":"bfdfe7dc352907fc980b868725387e989da425bb6d6d5d8dbb95e4afffb82217","name":"Flutter"},{"guid":"bfdfe7dc352907fc980b868725387e9890f568fc9b811cd56b08e401eec0f35e","name":"rive_common-rive_common_privacy"}],"guid":"bfdfe7dc352907fc980b868725387e9849c5bdbe31e3b466d37bab9271baf60a","name":"rive_common","predominantSourceCodeLanguage":"Xcode.SourceCodeLanguage.C-Plus-Plus","productReference":{"guid":"bfdfe7dc352907fc980b868725387e9884d4dff26bf6b3e66f738069c7476896","name":"rive_common.framework","type":"product"},"productTypeIdentifier":"com.apple.product-type.framework","provisioningSourceData":[{"bundleIdentifierFromInfoPlist":"${PRODUCT_BUNDLE_IDENTIFIER}","configurationName":"Debug","provisioningStyle":1},{"bundleIdentifierFromInfoPlist":"${PRODUCT_BUNDLE_IDENTIFIER}","configurationName":"Profile","provisioningStyle":1},{"bundleIdentifierFromInfoPlist":"${PRODUCT_BUNDLE_IDENTIFIER}","configurationName":"Release","provisioningStyle":1}],"type":"standard"}
```
Comment (jossean-rivera) [inline ios/build/ios/XCBuildData/PIFCache/target/TARGET@v11_hash=39047dbfca6f81dc03f9b20d0e79586f-json:1]: Are all these files required?

Comment (jossean-rivera) [reply ios/build/ios/XCBuildData/PIFCache/target/TARGET@v11_hash=39047dbfca6f81dc03f9b20d0e79586f-json]: It runs fine, I think it should be same.
Screenshot 2024-12-01 at 10 46 26 PM

**Comment by jossean-rivera** [comment]
I resolved the conflicts with main.

**Comment by jossean-rivera** [comment]
Builds fine even after removing the files. 

[Revert "Changes suggested in the pull request"](https://github.com/jossean-rivera/join_play/pull/17/commits/15565ccb114b4299a294ffdbfc7f1f605c560b06) 
[15565cc](https://github.com/jossean-rivera/join_play/pull/17/commits/15565ccb114b4299a294ffdbfc7f1f605c560b06)
[Deleted unnecessary changes](https://github.com/jossean-rivera/join_play/pull/17/commits/5ef6a0ba1e81661b781772ba822ad08c8294853f)
Screenshot 2024-12-01 at 10 59 49 PM

**Review by jossean-rivera — APPROVED** [review]

**Pull request merged**